aws/resource_aws_launch_template.go

```diff
@@ -800,60 +814,45 @@ func resourceAwsLaunchTemplateRead(d *schema.ResourceData, meta interface{}) err
 func resourceAwsLaunchTemplateUpdate(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).ec2conn
 
-	launchTemplateData, err := buildLaunchTemplateData(d)
-	if err != nil {
-		return err
-	}
+	version := strconv.Itoa(d.Get("default_version").(int))
 
-	dltv, err := conn.DescribeLaunchTemplateVersions(&ec2.DescribeLaunchTemplateVersionsInput{
-		LaunchTemplateId: aws.String(d.Id()),
-		Versions:         aws.StringSlice([]string{"$Latest"}),
-	})
-	if err != nil {
-		return fmt.Errorf("Error describing $Latest Launch Template Version for Launch Template(%s): %w", d.Id(), err)
-	}
-	if dltv == nil || len(dltv.LaunchTemplateVersions) == 0 {
-		return fmt.Errorf("Error describing $Latest Launch Template Version for Launch Template(%s): empty output", d.Id())
-	}
-
-	// Create new version only if content has changed since previous version
-	var defaultVersion string
-	description := d.Get("description").(string)
-	latestVersionDescription := aws.StringValue(dltv.LaunchTemplateVersions[0].VersionDescription)
+	if d.HasChanges(updateKeys...) {
+		launchTemplateData, err := buildLaunchTemplateData(d)
+		if err != nil {
+			return err
+		}
 
-	if (launchTemplateData.String() != dltv.LaunchTemplateVersions[0].LaunchTemplateData.String()) || (description != latestVersionDescription) {
 		launchTemplateVersionOpts := &ec2.CreateLaunchTemplateVersionInput{
 			ClientToken:        aws.String(resource.UniqueId()),
 			LaunchTemplateId:   aws.String(d.Id()),
 			LaunchTemplateData: launchTemplateData,
 		}
 
-		if description != "" {
-			launchTemplateVersionOpts.VersionDescription = aws.String(description)
+		if v, ok := d.GetOk("description"); ok {
+			launchTemplateVersionOpts.VersionDescription = aws.String(v.(string))
 		}
 
 		output, err := conn.CreateLaunchTemplateVersion(launchTemplateVersionOpts)
 		if err != nil {
-			return fmt.Errorf("Error creating new Launch Template Version for Launch Template(%s): %w", d.Id(), err)
+			return fmt.Errorf("error creating Launch Template (%s) Version: %w", d.Id(), err)
 		}
 
-		if v, ok := d.GetOk("update_default_version"); ok && v.(bool) {
-			defaultVersion = strconv.FormatInt(aws.Int64Value(output.LaunchTemplateVersion.VersionNumber), 10)
+		if output == nil || output.LaunchTemplateVersion == nil || output.LaunchTemplateVersion.VersionNumber == nil {
+			return fmt.Errorf("error creating Launch Template (%s) Version: empty output", d.Id())
 		}
-	}
 
-	if d.HasChange("default_version") {
-		defaultVersion = strconv.Itoa(d.Get("default_version").(int))
+		version = strconv.FormatInt(aws.Int64Value(output.LaunchTemplateVersion.VersionNumber), 10)
+
 	}
 
-	if defaultVersion != "" {
+	if d.Get("update_default_version").(bool) || d.HasChange("default_version") {
```

Comment:
@bflad should we also handle the case where `update_default_version` is updated to true but nothing in the templateData has changed e.g.
 `d.Get("update_default_version").(bool) && (version != d.Get("default_version).(string))`

Reply:
Sounds like a great case to also handle!